backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/bpartner/service/BPartnerPrintFormatRepository.java

```diff
@@ -87,21 +103,35 @@ public BPPrintFormat save(@NonNull final BPPrintFormat bpPrintFormat)
 
 	private I_C_BP_PrintFormat createOrUpdateRecord(@NonNull final BPPrintFormat bpPrintFormat)
 	{
-		final I_C_BP_PrintFormat bpPrintFormatRcord;
+		final I_C_BP_PrintFormat bpPrintFormatRecord;
 		if (bpPrintFormat.getBpPrintFormatId() > 0)
 		{
-			bpPrintFormatRcord = load(bpPrintFormat.getBpPrintFormatId(), I_C_BP_PrintFormat.class);
+			bpPrintFormatRecord = load(bpPrintFormat.getBpPrintFormatId(), I_C_BP_PrintFormat.class);
 		}
 		else
 		{
-			bpPrintFormatRcord = newInstance(I_C_BP_PrintFormat.class);
+			bpPrintFormatRecord = newInstance(I_C_BP_PrintFormat.class);
 		}
 
-		bpPrintFormatRcord.setC_BPartner_ID(bpPrintFormat.getBpartnerId().getRepoId());
-		bpPrintFormatRcord.setC_DocType_ID(bpPrintFormat.getDocTypeId());
-		bpPrintFormatRcord.setAD_Table_ID(bpPrintFormat.getAdTableId());
-		bpPrintFormatRcord.setAD_PrintFormat_ID(bpPrintFormat.getPrintFormatId());
+		bpPrintFormatRecord.setC_BPartner_ID(bpPrintFormat.getBpartnerId().getRepoId());
+		bpPrintFormatRecord.setC_DocType_ID(bpPrintFormat.getDocTypeId() == null ? 0 : bpPrintFormat.getDocTypeId().getRepoId());
+		bpPrintFormatRecord.setAD_Table_ID(bpPrintFormat.getAdTableId() == null ? 0 : bpPrintFormat.getAdTableId().getRepoId());
+		bpPrintFormatRecord.setAD_PrintFormat_ID(bpPrintFormat.getPrintFormatId() == null ? 0 : bpPrintFormat.getPrintFormatId().getRepoId());
+		bpPrintFormatRecord.setC_BPartner_Location_ID(bpPrintFormat.getBPartnerLocationId() == null ? 0 : bpPrintFormat.getBPartnerLocationId().getRepoId());
+		bpPrintFormatRecord.setDocumentCopies_Override(bpPrintFormat.getPrintCopies().toInt());
+
+		return bpPrintFormatRecord;
+	}
+
+	public SeqNo getNextSeqNo(@NonNull final BPartnerId bPartnerId)
+	{
+		final int lastLineInt = queryBL.createQueryBuilder(I_C_BP_PrintFormat.class)
+				.addOnlyActiveRecordsFilter()
+				.addEqualsFilter(I_C_BP_PrintFormat.COLUMNNAME_C_BPartner_ID, bPartnerId)
+				.create()
+				.maxInt(I_C_BP_PrintFormat.COLUMNNAME_SeqNo);
 
-		return bpPrintFormatRcord;
+		final SeqNo lastLineNo = SeqNo.ofInt(Math.max(lastLineInt, 10));
```

Comment:
it shall be max of lastLineInt or ZERO (and not ten)

Reply:
fixed